ext/travisci/prep-os-essentials-for

```diff
@@ -11,6 +11,21 @@ misuse() {
     exit 2
 }
 
+# this handles the issue where apt-get update fails but has an exit code of zero
+# because there's a mirror sync going on. This failure is retryable, but you have
+# to look for it in the output from the command. This is a common failure on AWS
+# and GCP because they inject their own mirrors into /etc/apt/sources.list at VM
+# start time, and there's a delay in getting all their mirrors in sync.
+apt_get_update() {
+    for i in $(seq 1 20); do
+        sudo apt-get update 2>&1 | tee /tmp/apt-get-update
+        set +e
+        grep -x "[WE]:.*" /tmp/apt-get-update; [ $? -ne 0  ] && break || sleep 1;
+    done;
+    set -e
+    echo Executed apt-get update $i times;
+}
```

Comment:
(TIL `grep -x`)
I might be tempted to be change this to something like:
```
for ...
    ...
    grep -x "[WE]:.*" "$tmpfile" || break
    sleep 1
done
echo ...
return 1  # Doesn't mattter, but suppose this could also be the failed apt-get exit status...
```